cabal-testsuite/PackageTests/ConditionalAndImport/cabal.test.hs

```diff
@@ -258,6 +258,41 @@ main = cabalTest . withRepo "repo" . recordMode RecordMarked $ do
   log "checking if we detect when the same config is imported via many different paths (we don't)"
   yopping <- cabal' "v2-build" [ "--project-file=yops-0.project" ]
 
+  log "checking \"using config from message\" without URI imports"
```

Comment:
This test is already kind of huge and unreadable, so I'm not a fan of making it even bigger, but I recognize that you didn't write it. Let's mark this for cleanup in the future though.

Reply:
I've removed this test.

Reply:
> Let's mark this for cleanup in the future though.

I can do this separately to this pull request.

Reply:
> This test is already kind of huge and unreadable

I did write this test and wanted to exercise the import feature a little more. Maybe I went a bit overboard with it? The project import feature was poorly tested when I started working on it.

```
$ git checkout cabal-install-v3.10.1.0

$ grep -R -E 'import:' cabal-testsuite/**/*.project
cabal-testsuite/PackageTests/ConditionalAndImport/cabal-cyclical.project:import: cabal-cyclical.project
cabal-testsuite/PackageTests/ConditionalAndImport/cabal.project:import: extra.project
cabal-testsuite/PackageTests/RelativePathProjectImports/cabal.project:import: ./dep/cabal.project
```